### ISSUE TITLE
Functional groups don't always match fuzzy traits

### ISSUE THREAD

**Issue by adletaw** [SrivastavaLab/bwg_names#24]
This one probably requires a discussion between @DEZERALDOlivier  and @dsrivast.

@DEZERALDOlivier did a lot of work filling in the functional groups column.  However, the `FG` columns in fuzzy traits should match up with the functional traits - and in several cases, they do not.  @aammd has written a [detailed report](https://htmlpreview.github.io/?https://raw.githubusercontent.com/SrivastavaLab/bwg_names/master/report/report.html) explaining the mismatches.  Take a look in the section labelled `functional group`.

In general, there are several inconsistencies in how the values in `FG1` through `FG5` translate into the 6 possible functional groups.

**Comment by DEZERALDOlivier** [comment]
Hi guys, thanks for taking the time to put all this work together. Inconsistencies between fuzzy traits and functional groups are not mistakes from me they are rather personal observations. For instance, chironomids, in fuzzy traits, are considered filter feeders while they are, I think, more gatherers. We can argue about it since chironomids are highly diversified, and within this taxonomic group I guess some are more filter feeders, gatherers or even predators (Tanypodinae). However, at the community level they may be more gatherers compared to culicids, what do you think? This was my rationale but I can be wrong let me know and I will fix this!

**Comment by aammd** [comment]
I want to comment on what I think is a related issue: what is represented by rows in this table which are not identified to species?

*  (A) Is each such row a different morphospecies, which we hope will turn out to be a single species upon closer testing? *Or* 

*  (B) Is an incompletely identified row meant to stand in for multiple unidentified species, in effect saying "This is all we know about the ecology of any species fitting this broad taxonomic description" ?

I was thinking about this mostly in relationship to @SrivastavaLab/bwgtools#10 (ie cases where researchers identified two different columns in the dataset to the same morphospecies). Does this happen because researchers felt that two distinct, unidentifiable specimens were both adequately described by the same broad category (B)?

Similarly, when we are debating fuzzy traits and functional groups, are we deciding what a **specific** animal does, or what, as you say, a group does "at the community level"?  It seems to me that if we go with the former we should be including NA a lot more often in this column.

tl;dr: Perhaps we need a

**Comment by adletaw** [comment]
In my opinion, each row should stand for a morphospecies that we hope to one day identify to the species level.  This is the easier way to deal with the data once it is in the database because it is hard to split up rows once they have been created.  This is also how the data sets have already been organized (as far as I know).  Each row is a morphospecies so some rows may have duplicate taxonomic information but different nicknames.  For example, the [two Scirtes species](https://github.com/SrivastavaLab/bwg_names/blob/master/data/Distributions_organisms_full.tsv#L66-L67).

**Comment by aammd** [comment]
@adletaw I completely agree.  However, i worried that another interpretation might be common when i saw that **two** research groups had independently used duplicate BWG nicknames in their precipitation experiment datasheets. Either these are errors, or we need a better articulation of what exactly this dataset (that is, `bwg_names`) actually means

**Comment by adletaw** [comment]
@aammd  Could it be that there is some resistance to adding new species?  This is definitely something that should be clarified for the purposes of populating the database in the future.  For your case, I wonder if people are looking at their species list and finding that two different morphospecies basically identify as one that already exists.  I think that means that the original morphospecies was either not described in enough detail, or that it encompasses two or more species because at the time of definition it was unknown that two species matched that description.  Or, as you said, it could be a mistake.  Sorry if I am misunderstanding your description of the problem.

I am definitely strongly against intentionally combining species into a single row.  That is just not going to be easy to fix if it is widespread.  I'm not exactly sure how this relates to Olivier's comments above.  My impression is that he assigned functional groups based on knowledge of a larger taxonomic group (e.g. chironomids), rather than based on knowledge of a particulary species (e.g. Chironomus detriticola).  Is this correct @DEZERALDOlivier?  Or did you take the opposite approach?  This relates to [another issue](https://github.com/SrivastavaLab/bwg_names/issues/22) in which @aammd suggested adding fuzzy traits for morphospecies based on knowledge about a higher level taxonomic group.  I'm not sure where I stand in that debate.  From a database point of view, it wouldn't be the end of the world to make changes to fuzzy traits for a particular species if we later decided they were wrong.

**Comment by DEZERALDOlivier** [comment]
I completely agree with @adletaw, I would not combine species into a single row, unless we are 100% sure they belong to the same species. My approach here was to assign functional groups based on knowledge of each particular species. For instance, Monopelopia (chironomid) is a predator. But based on what I know from several species sometimes fuzzy traits and functional groups did not match.

I think, for the sake of consistency, I will go back on the functional groups and will carefully follow fuzzy traits. I will make a decision if and only if similar values are in several fuzzy traits: 0-0-3-0-3-0.

What do you think?

**Comment by aammd** [comment]
@DEZERALDOlivier Well let's not go back and change that data just yet! Perhaps in some cases the fuzzy traits need to be re - examined, and in other cases it is the func.group. 
Have you had a chance to see the table at the end of the report? I was thinking that what we need is to see this table, but also some taxonomic information (so we know which species are involved)
/cc @dsrivast

**Comment by dsrivast** [comment]
Hi guys. @DEZERALDOlivier, @aammd @adletaw, There are a lot of intertwined issues here. (1) What to do when fuzzy traits and functional groups do not show a close match? This could occur for a number of reasons. One is that Olivier used his brain (always a good thing) and decided that chironomids and culicids cannot both be lumped together as filter feeders. They are subtly different, and that subtlety can be dealt with well by fuzzy traits but not a single functional grup category. So here, we should let Olivier's brain rule. The other reason is that the fuzzy traits are off. I currently have the summer students identifying inconsistencies in the fuzzy traits where they should all be the same (e.g. two morphospecies identified only to genus, that share the same genus) to deal with this, and hope to push commits on this when I am back from Costa Rica (away next week). (2) What is a morphospecies (row in table)? A morphospecies is something we think is a species. Rarely we have something that may be more than one species but we can't tell - like mosquito pupa that are very difficult to assign to larval stages. I wouldn't worry about this second category too much. The fuzzy traits for a morphospecies are gathered from what we now from the literature about organisms at the lowest level of taxonomic resolution that we could ID the morphospecies. So if Diptera.111 is identified as a Culex sp. (NOT CULEX SPP.) then I look up the traits for Culex larvae (all species in genera) and combine them into the fuzy trait codes. (3) What to do when two research groups used the same BWG nickname? This is often legitimate!! I can think of at least 3 species that occur in >1 drought expt site: Leptagrion andromache, Elpidium bromeliarum and Aulophorus superterraneus. Note that all are identified to the species level - that is why we can be sure that they are the same species in multiple sites. But, we should never have a morphospecies identified only at coarser resolution (e.g. genus) used in two sites because without DNA we on't know for sure this is the same. Could @adletaw look at the actual species pairs she noticed in multiple precipitation sites, and verify they are legit species-level ids? (4) What happens if we later lump species, or discover true duplicates? We should probably address this in the duplicates issue...we don't want duplicates, but I worry about how we would change the BWG codes in the raw survey/precip data to match the chosen BWG name.

**Comment by aammd** [comment]
@dsrivast  comments on each point in turn:

1. I agree that we should trust Olivier's brain. If I am interpreting your comment correctly, we will be focussing first on correcting fuzzy traits, and only _then_ revisit the functional groups (if necessary?)
   *  Regarding inconsistencies: ("I currently have the summer students identifying inconsistencies in the fuzzy traits where they should all be the same") Is this something we could query the table for?   Not sure if you've seen the final section of my report; some of the tables there could be extended to address this.

2. I agree that this was always my conception of what a row in the database represents. I'd suggest we write up a statement to this effect and post it somewhere, perhaps in the README here for the time being, and later on the front of the BWGdb

3. I apologize for the miscommunication. In two rainfall experiment datasets, two different research groups made the same "mistake" -- they named two columns with the same BWG nickname **in the same dataset**. There are two possibilites here:
  * typo
  * misconception: believing that a poorly-identified row in bwg_names represents ALL animals fitting that (broad) description. Rather, such a row represents a specific but incompletely-identifed animal.
See Srivastavalab/bwgtools#13 and Srivastavalab/bwgtools#10

4. Comments over at #21

**Comment by aammd** [comment]
The fuzzy trait columns and the functional group factor refer to different pieces of information. Organisms are not assigned to a functional group category by the fuzzy traits alone. The functional group represents a useful variable for filtering the dataset. the fuzzy traits are used in calculations of species diversity. 
The functional groups column also catches the consensus of the field/literature. (eg grouping mosquitoes as filter feeders)
